code/obj/machinery/computer/security/riot.dm

```diff
@@ -0,0 +1,116 @@
+		authed = 1
+		icon_state = "drawbr-alert"
+
+		for (var/obj/machinery/door/airlock/D in armory_area)
+			if (D.has_access(access_maxsec))
+				D.req_access = list(access_security)
+				D.no_access = 0
+			LAGCHECK(LAG_REALTIME)
+
+		if (armory_area)
+			for(var/obj/O in armory_area)
+				if (istype(O,/obj/storage/secure/crate))
+					O.req_access = list(access_security)
+				else if (istype(O,/obj/machinery/vending))
+					O.req_access = list(access_security)
+
+				LAGCHECK(LAG_REALTIME)
+
+
+	proc/print_auth_needed(var/mob/author)
+		if (author)
+			for (var/mob/O in hearers(src, null))
+				O.show_message("<span class='subtle'><span class='game say'><span class='name'>[src]</span> beeps, \"[author] request accepted. [src.auth_need - src.authorized.len] authorizations needed until shuttle is launched early.\"</span></span>", 2)
+		else
+			for (var/mob/O in hearers(src, null))
+				O.show_message("<span class='subtle'><span class='game say'><span class='name'>[src]</span> beeps, \"[src.auth_need - src.authorized.len] authorizations needed until shuttle is launched early.\"</span></span>", 2)
+
+
+/obj/machinery/computer/riotgear/attack_hand(mob/user as mob)
+	if (ishuman(user))
+		return src.attackby(user:wear_id, user)
+	..()
+
+//kinda copy paste from shuttle auth :)
+/obj/machinery/computer/riotgear/attackby(var/obj/item/W as obj, var/mob/user as mob)
+	interact_particle(user,src)
+	if(status & (BROKEN|NOPOWER))
+		return ..()
+	if (!user)
+		return ..()
+	if(authed)
+		boutput(user, "Armory has already been authorized!")
+		return
+
+	if (istype(W, /obj/item/device/pda2) && W:ID_card)
+		W = W:ID_card
+	if (!istype(W, /obj/item/card))
+		boutput(user, "No ID given.")
+		return ..()
+
+	if (!W:access) //no access
+		boutput(user, "The access level of [W] is not high enough.")
+		return
+
+	var/list/cardaccess = W:access
+	if(!istype(cardaccess, /list) || !cardaccess.len) //no access
+		boutput(user, "The access level of [W] is not high enough.")
+		return
+
+	if(!(access_security in W:access)) //doesn't have this access
+		boutput(user, "The access level of [W] is not high enough.")
+		return
+
+	if (!src.authorized)
+		src.authorized = list()
+
+	var/choice = alert(user, text("Would you like to (un)authorize access to riot gear? [] authorization\s are still needed.", src.auth_need - src.authorized.len), "Armory Auth", "Authorize", "Repeal")
+	if(get_dist(user, src) > 1) return
+	switch(choice)
+		if("Authorize")
+			if (user in src.authorized)
+				boutput(user, "You have already authorized! [src.auth_need - src.authorized.len] authorizations from others are still needed.")
+				return
+			if (access_maxsec in W:access)
+				authorize()
+				src.authorized = null
+				return
+
+			src.authorized += user //authorize by USER, not by registered ID. prevent the captain from printing out 3 unique ID cards and getting in by themselves.
```

Comment:
Might be interesting if it went by DNA, so a changeling, after eating three officers, could open the armory. Also this seems to imply that three staffies with a single stolen ID can pop the armory open?

Reply:
lol oh yeah sharing an id - i'll have to fix this